### ISSUE TITLE
difference of models

### ISSUE THREAD

**Issue by xins981** [yzqin/s4g-release#11]
@yzqin yuzhe, the demo model is PointNet2_tcls.py and it seem differrent with paper proposed, rather, PointNet2.py is consistent. which model provide the paper resulet?
also, which is  the corresponding dataset among YCBScenes and YCBScenesContact?
hope your reply, thank you for your time!

**Comment by yzqin** [comment]
hi @xins981,

I was wondering taht what do you mean by `PointNet2_tcls.py` is different from the paper proposed? The model [here]([contact_model.pth](https://github.com/yzqin/s4g-release/blob/master/inference/trained_models/contact_model.pth) provides the paper result.

YCBScenesContact is the data used by `contact_model.pth` while YCBScenes is the data used by `curvature_model.pth`

**Comment by xins981** [comment]
@yzqin, I mean that PointNet2_tcls.py directly regress [rotation matrix(9-dim)](https://github.com/yzqin/s4g-release/blob/b684ade8fa32ddb55d03a6a36fcfe7f1dc81ca1c/inference/grasp_proposal/network_models/models/PointNet2_tcls.py#L87) ranther than 6-dim representation a=[a1, a2].
besides, why the [translation](https://github.com/yzqin/s4g-release/blob/b684ade8fa32ddb55d03a6a36fcfe7f1dc81ca1c/inference/grasp_proposal/network_models/models/PointNet2_tcls.py#L90) is 4-dim vector?

so, Is there more good perfomance above configuration than paper say in practice?

**Comment by yzqin** [comment]
Hello @xins981,

You raise an interesting point. While it's true that we only use the first 6 values of the rotation prediction to compute the rotation matrix, as shown [here](https://github.com/yzqin/s4g-release/blob/b684ade8fa32ddb55d03a6a36fcfe7f1dc81ca1c/inference/grasp_proposal/grasp_detector.py#L124), we predict 9 values instead of 6 purely for convenience. This approach allows us to easily adjust the code to evaluate performance differences between various rotation representations. If my memory serves me right, the 6-value rotation representation tends to yield superior performance.

Regarding translation, the network's output is actually a softmax classification guided by a one-hot vector, rather than a continuous translation value. Since the training data we generate is discrete, we lack dense supervision. The way we utilize this 4-dimension translation vector can be found [here](https://github.com/yzqin/s4g-release/blob/b684ade8fa32ddb55d03a6a36fcfe7f1dc81ca1c/inference/grasp_proposal/grasp_detector.py#L178).

**Comment by xins981** [comment]
ok, thanks for your reply!
